CONTRIBUTING.md

```diff
@@ -1,12 +1,100 @@
-# Contributing
+# Contributing to UMF (Unified Memory Framework)
+
+- [Opening new issues](#opening-new-issues)
+- [Submitting Pull Requests](#submitting-pull-requests)
+    - [Building and testing](#building-and-testing)
+    - [Code style](#code-style)
+    <!-- - [Naming convention](#naming-convention) -->
+    - [License](#license)
+    - [Sign your work](#sign-your-work)
+    - [Adding new dependency](#adding-new-dependency)
+    - [Extending public API](#extending-public-api)
+
+Below you'll find instructions on how to contribute to UMF, either with code changes
+or issues. All contributions are most welcome!
+
+## Opening new issues
+
+Please log bugs or suggestions as [GitHub issues](https://github.com/oneapi-src/unified-memory-framework/issues).
+
+When reporting a bug remember about the details, at least:
+- version of UMF (hash of a commit or a tag),
+- OS and kernel,
+- compiler.
+
+When opening a new issue you can pick a predefined type of issue. Please follow
+the template and fill in all the information. If your query doesn't match any of
+the proposed types, just pick a general issue with no template.
+
+## Submitting Pull Requests
+
+We take outside code contributions to UMF through GitHub pull requests.
+You must first create your own fork of the project and submit your changes to a branch.
+You can then raise a pull request targeting `oneapi-src/unified-memory-framework:main`.
+Please try to keep commits neat and in order - please squash your commits to
+include only relevant ones (no "fixes after review" or similar).
+
+Before your pull request is merged it *must* pass all jobs in the GitHub
+Actions workflows and *must* be reviewed by no less than two reviewers.
```

Comment:
Please add a section about the review. Add info that "codeowners" would get a notification when PR is created, but one could always add specific people if needed which should speed up the review process. Also add info that PR is merged when CI passes, all discussions are resolved and PR got at least two approves.